salt/beacons/__init__.py

```diff
@@ -199,13 +199,44 @@ def _update_enabled(self, name, enabled_value):
             else:
                 self.opts['beacons'][name].append({'enabled': enabled_value})
 
-    def list_beacons(self):
+    def _get_beacons(self,
+                     include_opts=True,
+                     include_pillar=True):
+        '''
+        Return the beacons data structure
+        '''
+        beacons = {}
+        if include_pillar:
+            pillar_beacons = self.opts.get('pillar', {}).get('beacons', {})
+            if not isinstance(pillar_beacons, dict):
+                raise ValueError('Beacons must be of type dict.')
+            beacons.update(pillar_beacons)
+        if include_opts:
+            opts_beacons = self.opts.get('beacons', {})
+            if not isinstance(opts_beacons, dict):
+                raise ValueError('Beacons must be of type dict.')
+            beacons.update(opts_beacons)
+        return beacons
+
+    def list_beacons(self, where=None):
         '''
         List the beacon items
+
+        where: Whether to include beacon data from, either opts
```

Comment:
The naming of this is very confusing to me. I think we may need to consider a more descriptive name and also state all of the possible arguments for this in the docstring.